### DISCUSSION TITLE
Add DoubleCmd to Flathub repository

### DISCUSSION THREAD

**Discussion by Francewhoa** [doublecmd/doublecmd#1486] [Suggestions]
Hello DoubleCmd enthusiasts :)

This is a suggestion to add DoubleCmd to this Flathug repository at https://flathub.org

If needed, both me and the Ubertus.Org team would be happy to contribute testing and documentation

--- --- --- --- --- --- --- --- --- --- --- --- --- --- --- ---

Below is the same suggestion as above. But with details if you're interested in those.

--- --- --- --- --- --- --- --- --- --- --- --- --- --- --- ---

Benefit for maintainers:

• All-in-one packages. You save a lot of time. Because Flatpak doesn't depend on outside packages. And Flatpak automatically finds what to bundle using deduplication.

• Single package for all Linux distributions. List of distributions at https://flatpak.org/setup/

• Rapid updates. Send your updates directly to all your users. Your users do not have to wait for package repositories to update their list.

• Strong security. Using automated sandbox. With optional override for users of Flatseal.

• Use any repository to your liking. Your own, the https://codeberg.org , https://download.opensuse.org , GitLab, GitHub, any others. Details at https://docs.flatpak.org/en/latest/publishing.html

• Distribute any type of applications with Flatpak. Including. But not limited to, system level, command line, desktop, utilities, or independent applications. When packaging the application, to set the access level, simply add one easy word to the Flatpak parameter "--filesystem=". Details about this are in this documentation at https://docs.flatpak.org/en/latest/sandbox-permissions-reference.html#filesystem-permissions

___• Optionally, users can override the above permission using this free, easy to use, and powerful Flatseal at https://flathub.org/apps/com.github.tchx84.Flatseal

• If you are full, as maintainer, you could easily delegate to anyone to your liking both the creation and the maintenance of the application for the Flathub repository. In other words, any volunteer can add an application to Flathub. It does not have to be the maintainer.

• Developer guide at https://docs.flatpak.org/en/latest/

• Free community support for maintainers: 

___• Forum at https://discourse.flathub.org/

___• Instant message at https://matrix.to/#/#flatpak:matrix.org

___• Mailing list at https://lists.freedesktop.org/mailman/listinfo/flatpak

--- --- --- --- --- --- --- --- --- --- --- --- --- --- --- ---

Benefit for users:

• Easy and quick installation of the latest free application version at https://flathub.org

• Automatically use any theme to your liking for all apps or selected app. Assuming the app supports theming. This is powered by Flatseal.

___•Steps at https://discourse.flathub.org/t/gtk4-apps-do-not-use-adwaita-dark/1327/2

___•Get Flatseal at https://flathub.org/apps/com.github.tchx84.Flatseal

• Integration with GNOME. You are able to easily install Flatpak applications using the GNOME you're familiar with. When needed, you can still install Flatpak app using https://flathub.org , Terminal, and many other options.

• Of all the packaging options available, Flatpak definitely has the strongest security. Because each application is sanboxed. While at the same time, for some application, the default permissions grant too much access. So users who need strong security need to learn to use Flatseal to adapt the permissions at https://flathub.org/apps/com.github.tchx84.Flatseal

• Freedom. You get more freedom with Flatpak. Because https://flathub.org is controlled by a friendly not-for-profit community of users. In comparison, Snap Store is controlled by the for-profit corporation Canonical. So in the future, under constant pressure from its secret shareholders, when Canonical decides to close Snap Store, you will be in trouble. Not-for-profit community value you the people above profit. There are lots of benefit for you associated with this. Usually, for-profit corporations value money above you, the people. There are lots of risk associated with this.

___• Sources about who controls Flatpak is under "Acknowledgements" title at https://flathub.org/about

___• Sources about who controls Snap Store at https://ubuntu.com/legal/terms-and-policies/snap-store-terms

• Powerful but easy permissions override using this free Flatseal at https://flathub.org/apps/com.github.tchx84.Flatseal

--- --- --- --- --- --- --- --- --- --- --- --- --- --- --- ---

Challenge with Flatpak:

• The first few applications you install use significantly more storage space. Because Flatpak needs to download runtime(s). But those runtimes are downloaded only once and shared with all other applications. This means that on the long term, after installing the first few applications, the storage space used no longer significantly increase. Also, most users with recent devices or with large storage are not affected by this challenge.

--- --- --- --- --- --- --- --- --- --- --- --- --- --- --- ---

Related ticket at https://github.com/doublecmd/doublecmd/issues/309

Cheers,

Francewhoa

**Comment by alexx2000** [comment]
Looks like an advertisement :) Sandbox is a main disadvantage of Flatpak in our case. DC is a file manager and it should have a same file system access as system file manager. If I publish DC at flathub then users will post "bugs" that something does not work (due sandbox). And I will have to tell them that this is a normal.

**Reply by AsciiWolf** [reply]
Allowing wide file system access is not a problem in Flatpak. You can set `--filesystem=home` or `--filesystem=host` in the Flatpak manifest file and users can even easily override the default sandbox permissions if they want.

Opening files with apps directly from the Flatpak can be a problem though (if your app does not support standard portals), but should be easy to solve.

**Comment by Francewhoa** [comment]
> Looks like an advertisement :)

Indeed. An advertisement/arguments you asked for though ;) I mean when you asked "Do you have any arguments?" about Flatpak in your comment at https://github.com/doublecmd/doublecmd/issues/309#issuecomment-1002843286 That was a long while ago though.

If someone is asking, I do not have any conflict of interest with Flatpak. I'm just a user of Flatpak and happy with it :)

------------

> DC is a file manager and it should have a same file system access as system file manager. 

With the latest Flatpak version, application creators and maintainers are able to distribute any type of applications with Flatpak. Including. But not limited to, **system level,** command line, desktop, utilities, or independent applications. To do so, when packaging the application, to set the access level, simply add one easy word to the Flatpak parameter `--filesystem=`. Details about this are in this documentation at https://docs.flatpak.org/en/latest/sandbox-permissions-reference.html#filesystem-permissions

For example, but not limited to:

• `host` to access all system files. This is the access DoubleCmd (DC) seems to need. So when packaging DC, if appropriate, simply add `--filesystem=host`

• `host-etc` to allow access only to files in the user's `/etc` folder and all sub-folders and files

• `home` to allow access only to the user's home folder and all sub-folders and files

• `/some/path` to allow access to an arbitrary path

• `~/some/path` to allow access to an arbitrary path relative to the user's home folder and all sub-folders and files

Any folder above automatically includes its sub-folders and sub-files. Easy.


If your DC needs to read, write, and create folders and files, simply end any packaging permissions path with `:create`. Details about this at https://docs.flatpak.org/en/latest/sandbox-permissions-reference.html#filesystem-permissions

-----

> If I publish DC at flathub then users will post "bugs" that something does not work (due sandbox). And I will have to tell them that this is a normal.

This is a frequent challenge for most application creators and maintainers. To resolve this challenge, if any user complains about your application's default permission, for your review and your decision, I suggest to simply redirect them to this free Flatseal at https://flathub.org/apps/com.github.tchx84.Flatseal Suggesting to users to configure appropriately their Flatseal. And educating them at that the Flatpak distribution, by default, as the same system permission as the other DC distribution.

With this Flatseal above, users can override any default permissions using this free, easy to use, and powerful Flatseal GUI. If a user does not set the appropriate customized Flatseal permissions, its her/his problem. Not yours. Anyhow, using Flatpak, it is very quick and easy to re-install a fresh clean package and fresh default permissions.

--------

My original comment above contained false information about system level permission. My bad. In the future, before I regurgitate false information about Flatpak, I'll ask for their source and double-check that source. Then compare my finding with other sources who are without conflict of interest. I updated accordingly my first comment above in this discussion.

**Comment by TheNicker** [comment]
Flatpak :-1: 
So far haven't got any good experience with it

**Comment by AsciiWolf** [comment]
Flatpak packaging would be great. Immutable distributions such as SteamOS or Fedora Silverblue are becoming more and more important and Flatpak is the primary (or in many cases only) software distribution platform there.
